src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -136,6 +139,19 @@ class HttpGoogleCloudStorageDAO(
     }
   }
 
+  override def getWorkspaces(userId: String): Seq[(WorkspaceName, String)] = {
+    val credential = getGroupServiceAccountCredential
+    val directory = new Directory.Builder(httpTransport, jsonFactory, credential).setApplicationName("firecloud:rawls").build()
+    val groupIds = directory.groups().list().setUserKey(userId).execute().getGroups.map(g => g.getEmail.stripPrefix(s"${groupsPrefix}-").stripSuffix(s"@${appsDomain}")).map(groupId => groupId.split("_", 2))
+
+    val workspaceNames = groupIds.map(tuple => WorkspaceName(tuple.head, tuple(1).splitAt(tuple(1).lastIndexOf("-"))._1)).distinct
+
+    //we already get the access levels when we retrieve the group names, but sorting out the highest level from that is messier
+    val workspaceTuples = workspaceNames.map(wsname => (wsname, WorkspaceAccessLevel.toCanonicalString(getMaximumAccessLevel(userId, wsname))))
```

Comment:
Actually I changed the implementation so you're only a member of your highest-access group, so the extra lookup shouldn't be necessary, I think